ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/helpers/C_TypeHelper.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * ******************************************************************************
+ *   Copyright 2014-2015 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+package com.spectralogic.ds3autogen.c.helpers;
+
+import com.spectralogic.ds3autogen.api.models.Ds3Element;
+import com.spectralogic.ds3autogen.c.models.C_Type;
+import com.spectralogic.ds3autogen.c.models.ComplexType;
+import com.spectralogic.ds3autogen.c.models.PrimitiveType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.text.ParseException;
+import java.util.Set;
+
+public final class C_TypeHelper {
+    private static final Logger LOG = LoggerFactory.getLogger(C_TypeHelper.class);
+    private C_TypeHelper() {}
+
+    private static C_Type createType(final String type, final boolean isArray, final Set<String> enumNames) {
```

Comment:
That would work here, I'll update.  At first I didn't realize you could pass a Set as a parameter to a function that requires an ImmutableSet